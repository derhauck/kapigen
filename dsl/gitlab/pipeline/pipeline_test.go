package pipeline

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestNewDefaultCiPipeline(t *testing.T) {
	t.Run("create default pipeline", func(t *testing.T) {
		ciPipeline := (&CiPipeline{}).DefaultCiPipeline()
		stages := ciPipeline.Stages.Get()
		snaps.MatchSnapshot(t, ciPipeline.Default, ciPipeline.Workflow, ciPipeline.Variables, len(stages))
	})

}
