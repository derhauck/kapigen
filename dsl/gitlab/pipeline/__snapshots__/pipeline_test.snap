
[TestNewDefaultCiPipeline/create_default_pipeline - 1]
(*pipeline.CiPipelineDefault)(nil)
&pipeline.CiPipelineWorkflow{
    Name:  "default",
    Rules: &job.Rules{
        &job.Rule{
            If:           "$CI_MERGE_REQUEST_ID",
            Changes:      wrapper.Array[string]{},
            AllowFailure: false,
            Variables:    struct {}{},
            When:         job.WhenWrapper{
                Value: &enum.When(2),
            },
        },
        &job.Rule{
            If:           "$CI_DEFAULT_BRANCH == $CI_COMMIT_BRANCH",
            Changes:      wrapper.Array[string]{},
            AllowFailure: false,
            Variables:    struct {}{},
            When:         job.WhenWrapper{
                Value: &enum.When(2),
            },
        },
        &job.Rule{
            If:           "$CI_MERGE_REQUEST_IID && $CI_PIPELINE_SOURCE == 'merge_request_event'",
            Changes:      wrapper.Array[string]{},
            AllowFailure: false,
            Variables:    struct {}{},
            When:         job.WhenWrapper{
                Value: &enum.When(2),
            },
        },
        &job.Rule{
            If:           "$CI_COMMIT_TAG != null",
            Changes:      wrapper.Array[string]{},
            AllowFailure: false,
            Variables:    struct {}{},
            When:         job.WhenWrapper{
                Value: &enum.When(2),
            },
        },
    },
}
map[string]string{"KTC_STOP_PIPELINE":"false"}
int(8)
---
